frontend/src/tests/unit/storageManager.test.js

```diff
@@ -0,0 +1,82 @@
+import { getTimer, isStudent } from '../../utils/storageManager'
+
+describe('timer tests', () => {
+  const timeCases = [
+    { timeGiven: 1, timeExpected: '00:00:01' },
+    { timeGiven: 100, timeExpected: '00:01:40' },
+    { timeGiven: 0, timeExpected: '00:00:00' },
+    { timeGiven: 3600, timeExpected: '01:00:00' },
+    //maximum time we can show using this function
+    { timeGiven: 86399, timeExpected: '23:59:59' },
+    //times longer than 23:59:59 hours are cut off - the days moved to the left are not included in the result
+    { timeGiven: 9999999, timeExpected: '17:46:39' }
+  ]
+
+  it.each(timeCases)('Returns correct remaining time for %s', ({ timeGiven, timeExpected }) => {
+    //when
+    const remainingTime = getTimer(timeGiven)
+
+    //then
+    expect(remainingTime).toBe(timeExpected)
+  })
+})
+
+describe('isStudent tests', () => {
+  const studentTokens = [
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJqZ29yc2tpQHN0dWRlbnQuYWdoLmVkdS5wbCIsInJvbGVzIjpbIlNUVURFTlQiXSwiaXNzIjoiL2FwaS9sb2dpbiIsImV4cCI6MTY2NTMzNjQ4N30.JocsYBm7M_zj84WDsZh5TaM_pxCK9GFwCBre3gj6iTU'
+    },
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJtdXJiYW5za2FAc3R1ZGVudC5hZ2guZWR1LnBsIiwicm9sZXMiOlsiU1RVREVOVCJdLCJpc3MiOiIvYXBpL2xvZ2luIiwiZXhwIjoxNjY1MzM2NTU1fQ.rZOYxDKFmMwRYw_BXulP_l1sFS4KfuBoSL3610Gw7XY'
+    },
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJra3J1a0BzdHVkZW50LmFnaC5lZHUucGwiLCJyb2xlcyI6WyJTVFVERU5UIl0sImlzcyI6Ii9hcGkvbG9naW4iLCJleHAiOjE2NjUzMzY2MDZ9.QCSN8Nga6-tCZozbGCvkb4E_hbkE8nk_qy1wRs_MOtA'
+    },
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJzbWF6dXJAc3R1ZGVudC5hZ2guZWR1LnBsIiwicm9sZXMiOlsiU1RVREVOVCJdLCJpc3MiOiIvYXBpL2xvZ2luIiwiZXhwIjoxNjY1MzM2Njk1fQ.V9NOj0uHq8YMn66PUrWdMDFvyW2n_H2IbavGcwLVbv0'
+    }
+  ]
+  const professorTokens = [
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJibWFqQGFnaC5lZHUucGwiLCJyb2xlcyI6WyJQUk9GRVNTT1IiXSwiaXNzIjoiL2FwaS9sb2dpbiIsImV4cCI6MTY2NTMzNzA3NX0.M5sSb6c2hUw5Pfu1h1dZ8tnB7S3ESV8s2P0oAkX2zcI'
+    },
+    {
+      access_token:
+        'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJzemllbGluc2tpQGFnaC5lZHUucGwiLCJyb2xlcyI6WyJQUk9GRVNTT1IiXSwiaXNzIjoiL2FwaS9sb2dpbiIsImV4cCI6MTY2NTMzNzExNX0.ZKcLZ2ARHtA5Q1TulX9ow0C5KGnV0KxILC8VvJ4P2sk'
+    }
+  ]
+  const invalidTokens = [
+    { access_token: '' },
+    { access_token: 'incorrectToken' },
+    { access_token: '111111111111111111111111111111111111111111111111111' }
+  ]
```

Comment:
Zgadzam się niestety z guardianem, nie podoba mi się to 🫤
Ale nie bój żaby - przychodzę z rozwiązaniem 😆 
1. Zainstaluj bibliotekę 'jsonwebtoken':
 `npm i jsonwebtoken`
2. Zaimportuj ją do pliku z testami
`import jwt from 'jsonwebtoken'`
3. Wywal `studentTokens` i `professorTokens`
4. Zamiast powyżej wymienionych list zrób listy danych zakodowanych w tokenie w taki sposób (po jednym przykładzie, więcej dopisz sam):
```jsx
  const students = [{ sub: 'test@student.agh.edu.pl', roles: [AccountType.STUDENT] }]
  const professors = [{ sub: 'prof@agh.edu.pl', roles: [AccountType.PROFESSOR] }]
```
5. Żeby z powyższych danych wygenerować token za pomocą biblioteki robisz wewnątrz testu coś takieg:
```jsx
const token = jwt.sign(student, 'secret', { expiresIn: '1h' })
```
W tym momencie masz token, którego możesz wrzucić do funkcji helpera ;)

Reply:
PS. invalidTokens są raczej git, te testy można zostawić ;)

Reply:
No spoko, ale to nam nie przetestuje tego na "prawdziwych" danych 🤔

Reply:
Poza tym - guardian nie czepia się tych tokenów, tylko tego jednego "niepełnego", bo odczytał go jako nieposiadający sekretu na koniec - tzn niezmienny, więc nie powinien być taki w kodzie

Reply:
Ogólnie może i to jest trochę "brzydkie", ale generowanie swoich tokenów na froncie i testowanie ich mija się z celem, ta libka wygląda jak coś co wpina się do expressa jak robisz back na node.js... przynajmniej taka jest moja opinia

Reply:
jak uważasz